crates/astria-proto/proto/astria/execution/v1alpha2/execution.proto

```diff
@@ -0,0 +1,107 @@
+syntax = 'proto3';
+
+package astria.execution.v1alpha2;
+
+import "google/protobuf/timestamp.proto";
+
+// The set of information which deterministic driver may to know about a given executed Block
+message Block {
+  // The block number
+  uint32 number = 1;
+  // The hash of the block
+  bytes hash = 2;
+  // Timestamp on the block, standardized to google protobuf standard.
+  google.protobuf.Timestamp timestamp = 3;
+}
+
+// These fields should be indexed on most block chains, and can be used to identify a block.
+message BlockIdentifier {
+  oneof identifier {
+    uint32 block_number = 1;
+    bytes block_hash = 2;
+  }
+}
+
+message GetBlockRequest {
+  BlockIdentifier identifier = 1;
+}
+message GetBlockResponse {
+    Block block = 1;
+}
+
+message BatchGetBlocksRequest {
+  repeated BlockIdentifier identifiers = 1;
+}
+message BatchGetBlocksResponse {
+  repeated Block blocks = 1;
+}
+
+
+// CreateBlockRequest contains all the information needed to create a new executed block.
+//
+// This information comes from previous execution blocks, as well as from sequencer blocks.
+message CreateBlockRequest {
+  // The hash of the block which will be executed on top of
+  bytes prev_block_hash = 1;
+  // List of transactions to include in the new block
+  repeated bytes transactions = 2;
+  // The timestamp which should be used for new block
+  google.protobuf.Timestamp timestamp = 3;
+}
+// CreateBlockResponse is returned after calling CreateBlock
+message CreateBlockResponse {
+  // The executed block which was created.
+  Block block = 1;
+}
+
+// The CommitmentState holds the block at each stage of sequencer commitment level
+//
+// A Valid CommitmentState:
+// - Block numbers are such that soft+1 >= head >= soft >= firm.
```

Comment:
```suggestion
// - Block numbers are such that head >= soft >= firm.
```

soft+1 can never be more than head? unless head is already softly committed (which shouldn't be the case with tendermint)

Reply:
okay, i see `head` can equal `soft` now. however, i find this a bit confusing since `soft+1 >= head` implies `head < soft+1` which is not always the case, `head == soft+1` can occur and probably often will

Reply:
Basically the options are only soft + 1 or soft, I think this accurately represents that? Perhaps it would be better split out into a seperate validation list so it's `head >= soft >= firm` and `head block never be more than 1 block ahead of soft block`?

Reply:
Does this make more sense?
`head >= (safe + 1 or safe) >= firm` with:
`head >= safe + 1` when `execution_commitment_level` == `head`
 and `head >= safe` when `execution_commitment_level` == `soft`